clean/hammond_pd_cprr.py

```diff
@@ -0,0 +1,149 @@
+        .str.replace(' / ', ' | ', regex=False)
+    return df.drop(columns='final_recommendation')
+
+
+def combine_columns(df):
+    def combine(row):
+        txts = []
+        if pd.notnull(row.violation_1):
+            txts.append('%s' % row.violation_1)
+        if pd.notnull(row.violation_2_3):
+            txts.append('%s' % row.violation_2_3)
+        return '| '.join(txts)
+    df.loc[:, 'charges'] = df.apply(combine, axis=1, result_type='reduce')
+    df = df.drop(columns=['violation_1', 'violation_2_3'])
+    return df
+
+
+def clean_charges(df):
+    df.charges = df.charges.str.lower().str.strip().fillna('')\
+        .str.replace(r'^(\d+) (\w+)', r'\1 - \2', regex=True)\
+        .str.replace('.', '', regex=False)\
+        .str.replace(r' \bgo\b ', ' ', regex=True)\
+        .str.replace('/', ' and ', regex=False)\
+        .str.replace(r'\(founded\)', '', regex=True)\
+        .str.replace(r'^excessive force ', 'excessive force | ', regex=True)\
+        .str.replace('omissionn', 'omission', regex=False)\
+        .str.replace(r' ?(unbec?o?i?mo?ing) ?( ?(of)? ?(an officer)?)? ?', ' unbecoming ', regex=True)\
+        .str.replace('video conduct unbecoming', 'video | conduct unbecoming', regex=False)
+    return df
+
+
+def combine_duplicate_action_disposition_and_charges_rows(df):
+    df = df.groupby(['uid', 'tracking_number']).agg({
+        'first_name': 'first',
+        'last_name': 'first',
+        'department_desc': 'first',
+        'disposition': ''.join, 
+        'action': ''.join,
+        'charges': ' |'.join,
+        'incident_year': 'first',
+        'incident_month': 'first',
+        'incident_day': 'first',
+        'investigation_start_year': 'first',
+        'investigation_start_month': 'first',
+        'investigation_start_day': 'first',
+        'data_production_year': 'first',
+        'agency': 'first',
+        'complaint_uid': 'first'
+        }).reset_index()
+    return df
+
+
+def drop_rows_without_tracking_number(df):
+    return df[df.tracking_number != ''].reset_index(drop=True)
+
+
+def clean():
+    df = pd.read_csv(data_file_path('hammond_pd/hammond_pd_cprr_2015_2020.csv'))\
+        .pipe(clean_column_names)
+    df = df\
+        .pipe(split_name)\
+        .pipe(clean_tracking_number)\
+        .pipe(clean_incident_date)\
+        .pipe(clean_disposition)\
+        .pipe(clean_action)\
+        .pipe(combine_columns)\
+        .pipe(clean_charges)\
+        .pipe(clean_department_desc)\
+        .pipe(clean_dates, ['incident_date', 'investigation_start_date'])\
+        .pipe(drop_rows_without_tracking_number)\
+        .pipe(standardize_desc_cols, ['department_desc', 'action', 'charges'])\
+        .pipe(set_values, {
+            'agency': 'Hammond PD',
+            'data_production_year': '2021'
+        })\
+        .pipe(gen_uid, ['first_name', 'last_name', 'agency'])\
+        .pipe(gen_uid, 
+        ['uid', 'charges', 'tracking_number', 'disposition'], 'complaint_uid')\

```

Comment:
Since you're compressing the rows down to one row per `tracking_number` x `uid` I doubt that we need to include `charges` and `disposition` here. But then I think splitting the rows out to 1 row per charge is better and more consistent.